### PR TITLE
support diff or abs = 0, meaning disable them.

### PR DESCRIPTION
**Pull request by Jun10ng**

### CODE REVIEW

**Comment by cch123** [comment]
maybe provide a const value naming disable is better:

```go
package holmes
const DISABLED

package usermain


```

File: readme.md [hunk 1]
```diff
@@ -72,7 +72,7 @@ h.Stop()
 * WithGoroutineDump(10, 25, 2000, 100*1000,time.Minute) means dump will happen when current_goroutine_num > 10 && 
   current_goroutine_num < `100*1000` && current_goroutine_num > `125%` * previous_average_goroutine_num or current_goroutine_num > `2000`,
   `time.Minute` means once a dump happened, the next dump will not happen before cooldown
-  finish-1 minute.
+  finish-1 minute, if `abs` and `diff` is 0 meaning disable it, if `abs` and `diff` is 0 meaning disable it.
```
Comment (doujiang24) [inline readme.md:75]: duplicated?

File: util.go [hunk 2]
```diff
@@ -179,14 +179,16 @@ func matchRule(history ring, curVal, ruleMin, ruleAbs, ruleDiff, ruleMax int) (b
 	}
 
 	// the current peak load exceed the absolute value
-	if curVal > ruleAbs {
+	if curVal > ruleAbs && ruleAbs!=0{
```
Comment (doujiang24) [inline util.go:182]: it's better to add a new condition at the front of this function, that will make the code simpler.

**Review by doujiang24 — COMMENTED** [review]
Also, agree with @cch123 , a const value could be better.

**Pull request closed without merging**